### PR TITLE
Node-Before-After files to be added to the codebase

### PR DESCRIPTION
**Pull request by shivanimall**

### CODE REVIEW

File: vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml [hunk 1]
```diff
@@ -263,10 +263,10 @@
 
     <!-- op:is-same-node($parameter1 as node(), $parameter2 as node()) as xs:boolean -->
     <operator name="op:is-same-node">
-        <param name="parameter1" type="node()?"/>
-        <param name="parameter2" type="node()?"/>
+        <param name="parameter1" type="node()"/>
+        <param name="parameter2" type="node()"/>
         <return type="xs:boolean"/>
-    </operator>
+     </operator>
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/functions/builtin-operators.xml]: Why the extra space?

File: vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/node/AbstractNodePositionalCheckEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    private final NodeTreePointable ntp1 = (NodeTreePointable) NodeTreePointable.FACTORY.createPointable();
+    private final NodeTreePointable ntp2 = (NodeTreePointable) NodeTreePointable.FACTORY.createPointable();
+    private final TypedPointables tp = new TypedPointables();
+
+    @Override
+    protected void evaluate(TaggedValuePointable[] args, IPointable result) throws SystemException {
+
+        parameterTypeCheck(args[0], ntp1);
```
Comment (prestoncarman) [inline vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/node/AbstractNodePositionalCheckEvaluator.java]: I think it makes the code cleaner and short if you don't have the function call and just have two if statements like the following:

if (args[0].getTag() != ValueTag.NODE_TREE_TAG) {
    throw new SystemException(ErrorCode.FORG0006);
}

**Comment by prestoncarman** [comment]
Only two requests.

**Comment by shivanimall** [comment]
My bad. Previously made changes did not apply on this branch. Code updated. Thanks!

**Pull request merged**